addons/machinecontroller/deployment-controller.yaml

```diff
@@ -68,6 +68,9 @@ spec:
             {{ if .OperatingSystemManagerEnabled -}}
             - -use-osm
             {{ end -}}
+            {{ if .Config.CloudProvider.External }}
+            - -node-external-cloud-provider
+            {{ end }}
```

Comment:
It's provided to the webhook: https://github.com/kubermatic/kubeone/blob/eaffa27a5f62c7974f90eb62bd7594771016704f/addons/machinecontroller/webhook.yaml#L87-L89

Reply:
I remember adding this with https://github.com/kubermatic/kubeone/pull/1748. Anyways it does make sense to be consistent with the flags. Although, the webhook takes care of adding the required annotations so it should have worked fine.

Reply:
If I'm not mistaken, those flags on MC are noop. They're left to prevent breaking changes, with the intention to remove them in some later release. I'll double-check this tomorrow.

Reply:
exactly

Reply:
- https://github.com/kubermatic/machine-controller/blob/master/pkg/controller/machine/machine_controller.go#L766
- https://github.com/kubermatic/machine-controller/blob/master/pkg/userdata/amzn2/provider.go#L263
- https://github.com/kubermatic/machine-controller/blob/master/pkg/userdata/helper/kubelet.go#L162
- https://github.com/kubermatic/machine-controller/blob/master/pkg/userdata/helper/kubelet.go#L47

They are still used though. Btw the `--cloud-provider` flag was supposed to be removed with k8s 1.24 but they decided to opt out of it. https://github.com/kubernetes/kubernetes/issues/110018

Reply:
With regards to the first link, the value is obtained via:

* the command-line flag on MC
* the `v1.kubelet-flags.machine-controller.kubermatic.io/ExternalCloudProvider` annotation on the Machine object

In addition to that, the annotation has a priority over the command-line flag, making the command-line flag effectively noop. The annotation is applied by the webhook when creating a Machine object, based on flags provided to the webhook. In other words, we can provide that flag to MC as well, but it's redundant and we should eventually remove it from MC. To be able to remove the flag more easily, we should stop using that flag. That's why I wouldn't introduce it back again.

Reply:
> In addition to that, the annotation has a priority over the command-line flag

Yes, if the annotation is specified. But calling them "noop" is a bit far-fetched. Anyways as I said I can remove this since for us this flag in webhook takes care of it.

Reply:
a flag is no-op in the controller, the webhook does act on it.